File: Backend/Core.Domain/DocumentType.cs
using Core.Domain.Enum;
using Framework.Domain;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Core.Domain
{
    public class DocumentType : EntityBase
    {

        [Required(ErrorMessage = "O nome do tipo de documento é obrigatório")]
        public string Name { get; set; }

        [Required(ErrorMessage = "O tipo de pessoa é obrigatório")]
        [MaxLength(30)]
        public PersonType PersonType { get; set; }

        [Required(ErrorMessage = "O Id do país é obrigatório")]
        [ForeignKey(nameof(FKCountry))]
        public int IdCountry { get; set; }
        public Country FKCountry { get; set; }

    }
}

File: Frontend/src/app/system/_domains/DocumentType.ts
import { Injectable } from '@angular/core';
import {_EntityBase} from './_EntityBase'
import { Country } from './Country';
import { PersonType } from './enums/PersonType';

@Injectable()
export class DocumentType extends _EntityBase
{
     public Name: string;
     public PersonType: PersonType;
     public IdCountry: number;
     public FKCountry: Country;

}
